### ISSUE TITLE
Multi-threading of core module

### ISSUE THREAD

**Issue by IceFlovver** [amosproj/amos2022ws01-firmware-scraper#114]
## User story
1. As a user
2. I want to scrape multiple vendors at the same time
3. So that the scraping is time-efficient

## Acceptance criteria
* Vendor modules are executed in parallel

## Definition of done (DoD)
* All the acceptance criteria have been met
* The code has been merged with the main one
* The documentation and Bill of Materials have been updated
* The code is complete and uploaded in GitHub
* Functionality has been reviewed

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
